pkg/utils/slice/slice.go

```diff
@@ -24,10 +24,10 @@ func ContainsString(slice []string, s string) bool {
 	return false
 }
 
-// RemoveString returns a newly created []string that contains all items from slice that
+// Remove returns a newly created slice that contains all items from slice that
 // are not equal to s.
-func RemoveString(slice []string, s string) []string {
-	newSlice := make([]string, 0)
+func Remove[K comparable](slice []K, s K) []K {
+	newSlice := make([]K, 0)
```

Comment:
What about using standard library function `DeleteFunc()` ?
```suggestion
func Remove[K comparable](slice []K, s K) []K {
	return slices.DeleteFunc(slice, func(t K) bool {
		return t == s
	})
}
```